### PR TITLE
Bugfix: Correct calendar week date calculation logic

### PR DESCRIPTION
**Pull request by christoph-teichmeister**
Fixes a bug in `get_start_and_end_date_from_calendar_week` where the first calendar week was incorrectly determined. Adjusted the logic to compute the correct week start date based on ISO week rules. Added comprehensive test cases to ensure proper functionality for edge cases.

Bärbel and I stumbled upon this 🫠🫠🫠

### CODE REVIEW

File: ambient_toolbox/utils/date.py [hunk 1]
```diff
@@ -92,21 +92,35 @@ def datetime_format(target_datetime: datetime.datetime, dt_format: str) -> str:
...
+    first_day_of_year = datetime.date(year=year, month=1, day=1)
 
-        fourth_of_january = datetime.date(year=year, month=1, day=4)
+    # Determine which day of the week the first day of the year falls on
+    # (0 = Monday, 6 = Sunday)
+    weekday = first_day_of_year.weekday()
 
-        start_of_week = fourth_of_january - relativedelta(days=fourth_of_january.weekday())
+    # monday = 0, sunday = 6
+    thursday = 3
```
Comment (GitRon) [inline ambient_toolbox/utils/date.py:106]: suggestion: can we use a constant?

File: tests/test_utils_date.py [hunk 2]
```diff
@@ -46,6 +46,14 @@ def test_get_start_and_end_date_from_calendar_week(self):
         self.assertEqual(monday, datetime.date(year=2017, month=7, day=24))
         self.assertEqual(sunday, datetime.date(year=2017, month=7, day=30))
 
+        monday, sunday = get_start_and_end_date_from_calendar_week(2025, 3)
```
Comment (GitRon) [inline tests/test_utils_date.py:49]: question: do these test cases cover the changes? 🤔

File: ambient_toolbox/utils/date.py (hunk 1 above)
Comment (christoph-teichmeister) [reply ambient_toolbox/utils/date.py:106]: image


tried to, wasn't python 3.11 compatible :/

[Commit](https://github.com/ambient-innovation/ambient-toolbox/pull/43/commits/189a684b0bb62d55a9586572514736541c44c634) | [Job](https://github.com/ambient-innovation/ambient-toolbox/actions/runs/12828167696/job/35771631716?pr=43)

Comment (GitRon) [reply ambient_toolbox/utils/date.py:106]: Isn't it `calendar.THURSDAY`?

File: tests/test_utils_date.py (hunk 2 above)
Comment (christoph-teichmeister) [reply tests/test_utils_date.py:49]: yes, the third calendar week of 2025 is incorrectly resolved to the week starting with 20.01.2025, when using the old technique (`datetime.datetime.strptime(f"{year}-{calendar_week}-1", "%Y-%W-%w").astimezone().date()` => `datetime.datetime.strptime("2025-3-1", "%Y-%W-%w").astimezone().date()` => 20.01.2025), when it should be the 13.01.2025.

This is covered with this test, which asks for the third calendar week of 2025 :slightly_smiling_face:  

image

image

File: ambient_toolbox/utils/date.py (hunk 1 above)
Comment (christoph-teichmeister) [reply ambient_toolbox/utils/date.py]: aha! Messed up my imports 🤡

**Review by GitRon — APPROVED** [review]
